tools/antithesis-go-instrumentor/antithesis-go-instrumentor.go

```diff
@@ -0,0 +1,227 @@
+		}
+		hasher.Write(bytes)
+	}
+
+	return hex.EncodeToString(hasher.Sum(nil))[0:12]
+}
+
+func copyRecursiveNoClobber(from, to string) {
+	commandLine := fmt.Sprintf("cp --update=none --recursive %s/* %s", from, to)
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+	cmd := exec.Command("bash", "-c", commandLine)
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	logger.Printf("%q", commandLine)
+	err := cmd.Run()
+	if err != nil {
+		logger.Fatalf("%+v", err)
+	}
+}
+
+func addDependencies(customerInputDirectory, customerOutputDirectory string) {
+	commandLine := fmt.Sprintf("(cd %s; go mod edit -require=github.com/antithesishq/antithesis-sdk-go/instrumentation@latest -print > %s/go.mod)",
+		customerInputDirectory,
+		customerOutputDirectory)
+
+	cmd := exec.Command("bash", "-c", commandLine)
+	logger.Printf("%s", commandLine)
+	_, err := cmd.Output()
+	if err != nil {
+		// Errors here are pretty mysterious.
+		logger.Fatalf("%v", err)
+	}
+}
+
+func writeInstrumentedSource(source, path string) error {
+	// Any errors here are fatal anyway, so I'm not checking.
+	f, e := os.Create(path)
+	if e != nil {
+		logger.Printf("Error: could not create %s", path)
+		return e
+	}
+	defer f.Close()
+	_, e = f.WriteString(source)
+	if e != nil {
+		logger.Printf("Error: Could not write instrumented source to %s", path)
+		return e
+	}
+	return nil
+}
+
+func main() {
+	cmd_args := parse_args()
+	if cmd_args.ShowVersion {
+		fmt.Println(strings.TrimSpace(versionString))
+		os.Exit(0)
+	}
+
+	if cmd_args.InvalidArgs {
+		os.Exit(1)
+	}
+
+	var err error
+
+	// Setup logging globals
+	// TODO Could also pass these settings inside a parameter (or inside a receiver)
+	logger = cmd_args.Logger
+	verbosity = cmd_args.Verbosity
+
+	logger.Println(strings.TrimSpace(versionString))
+
+	if !IsGoAvailable() {
+		logger.Printf("Go toolchain not available")
+		os.Exit(1)
+	}
+
+	var cmd_files *CommandFiles
+
+	if err, cmd_files = cmd_args.NewCommandFiles(); err != nil {
+		logger.Printf(err.Error())
+		os.Exit(1)
+	}
+
+	if err = cmd_files.ParseExclusionsFile(); err != nil {
+		logger.Printf(err.Error())
+		os.Exit(1)
+	}
+
+	sourceFiles := []string{}
+	if err, sourceFiles = cmd_files.FindSourceCode(); err != nil {
+		logger.Printf(err.Error())
+		os.Exit(1)
+	}
+
+	files_hash := HashFileContent(sourceFiles)[0:12]
+	var instrumentor *Instrumentor = nil
+	var symbolTableWriter *SymbolTable = nil
+
+	if cmd_args.WantsInstrumentor {
+		logger.Printf("Instrumenting %s to %s", cmd_files.InputDirectory, cmd_files.CustomerDirectory)
+		symbolTableWriter = cmd_files.CreateSymbolTableWriter(files_hash)
+		instrumentor = CreateInstrumentor(cmd_files.InputDirectory, InstrumentationModuleName, symbolTableWriter)
+	}
+	usingSymbols := cmd_files.UsingSymbols()
+	full_catalog_path := cmd_files.CatalogPath
+
+	// Setup the assertion scanner (used to create the assertion catalog)
+	aSI := NewScanningInfo(verbosity > 0, full_catalog_path, usingSymbols)
```

Comment:
Could we call this variable something better, like ``catalogScanner`` or ``assertionScanner``?

Reply:
Leaving it as `aSI` and `cI` for now.  Here's the full context:

```
	//--------------------------------------------------------------------------------
	// Setup coverage and assertion processors
	//--------------------------------------------------------------------------------
	cI := cmd_files.NewCoverageInstrumentor()
	aSI := NewScanningInfo(verbosity > 0, cI.FullCatalogPath, cI.UsingSymbols)

	//--------------------------------------------------------------------------------
	// Process all files (ignore previously generated assertion catalogs)
	//--------------------------------------------------------------------------------
	for _, file_name := range source_files {
		if IsGeneratedFile(file_name) {
			logger.Printf("Skipping %s", file_name)
			continue
		}

		if instrumented_source := cI.InstrumentFile(file_name); instrumented_source != "" {
			cmd_files.WriteInstrumentedOutput(file_name, instrumented_source, cI)
		}

		aSI.ScanFile(file_name)
	}

	//--------------------------------------------------------------------------------
	// Wrap-up processing, summarize results in logger
	//--------------------------------------------------------------------------------
	cmd_files.WrapUp()
	edge_count := cI.WrapUp()
	cI.SummarizeWork(len(source_files))

	aSI.WriteAssertionCatalog(edge_count)
	aSI.SummarizeWork()
```